packages/lib-utils/src/app/AppInitSDK.tsx

```diff
@@ -0,0 +1,51 @@
+import { consoleLogger } from '@monorepo/common';
+import * as React from 'react';
+import { Provider } from 'react-redux';
+import type { UtilsConfig } from '../config';
+import { setUtilsConfig } from '../config';
+import type { InitAPIDiscovery } from '../types/api-discovery';
+import { initAPIDiscovery } from './api-discovery';
+import { useReduxStore } from './redux';
+
+type AppInitSDKProps = React.PropsWithChildren<{
+  configurations: {
+    apiDiscovery?: InitAPIDiscovery;
+    appFetch: UtilsConfig['appFetch'];
+  };
+}>;
+
+/**
+ * Component for providing store access to the SDK.
+ * Add this at app-level to make use of app's redux store and pass configurations prop needed to initialize the app, preferred to have it under Provider.
+ * It checks for store instance if present or not.
+ * If the store is there then the reference is persisted to be used in SDK else it creates a new store and passes it to the children with the provider
+ * @component AppInitSDK
+ * @example
+ * ```ts
+ * return (
+ *  <Provider store={store}>
+ *   <AppInitSDK configurations={{ appFetch, apiDiscovery }}>
+ *      <CustomApp />
+ *      ...
+ *   </AppInitSDK>
+ *  </Provider>
+ * )
+ * ```
+ */
+const AppInitSDK: React.FC<AppInitSDKProps> = ({ children, configurations }) => {
+  const { store, storeContextPresent } = useReduxStore();
+  React.useEffect(() => {
+    const { appFetch, apiDiscovery = initAPIDiscovery } = configurations;
```

Comment:
Destruct outside of `useEffect`. We want to key off the values of each, not the object itself.

`useEffect` uses referential equality and the object is not needed to be part of that check.

@vojtechszocs made [the request here](https://github.com/openshift/dynamic-plugin-sdk/pull/28#discussion_r809276108) too.

Reply:
@andrewballantyne apologies - this got lost in the lot of comments